src/main/java/com/exadel/discount/service/impl/FavoriteServiceImpl.java

```diff
@@ -83,39 +85,46 @@ public FavoriteDTO findFavoriteById(UUID id) {
 
     @Transactional
     @Override
-    public FavoriteDTO assignFavoriteToUser(CreateFavoriteDTO createFavoriteDTO) {
-        log.debug("Finding of certain user by ID");
+    public FavoriteDTO assignFavoriteToUser(UUID discountId) {
+        log.debug(String.format("Check: if user already have Favorite of Discount with ID %s ", discountId));
+        String userEmail = SecurityContextHolder.getContext().getAuthentication().getName();
+        if (favoriteRepository.findByDiscountIdAndAndUserEmail(discountId, userEmail).isPresent()) {
+            throw  new CreationRestrictedException(String
+                    .format("Favorite for discount with id %s does already exist", discountId));
+        }
```

Comment:
You can use existsByDiscountIdAndUserEmail, it'll return boolean

Reply:
the same